### PR TITLE
validation refactor and cloudprofile validation

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/priority normal
/platform azure

**What this PR does / why we need it**:
Refactors the azure validation to use gardener extension packages and adds `CloudProfile` validation.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Azure admission controller now validates `CloudProfile` creation/update.
```

### CODE REVIEW

**Comment by kon-angelo** [comment]
Currently set to draft, pending the ongoing release validation.

File: example/40-validatingwebhookconfiguration.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ webhooks:
     - UPDATE
     resources:
     - shoots
```
Comment (ialidzhikov) [inline example/40-validatingwebhookconfiguration.yaml:17]: You could also update the name of the example ValidatingWebhookConfiguration from `gardener-extension-validator-azure` to `gardener-extension-admission-azure`.

File: pkg/admission/validator/cloudprofile.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// Validate validates the given CloudProfile objects.
+func (cp *cloudProfile) Validate(ctx context.Context, new, old client.Object) error {
+	cloudProfile, ok := new.(*core.CloudProfile)
+	if !ok {
+		return fmt.Errorf("wrong object type %T", new)
+	}
+
+	providerConfigPath := field.NewPath("spec").Child("providerConfig")
+	if cloudProfile.Spec.ProviderConfig == nil {
+		return field.Required(providerConfigPath, "providerConfig must be set for AWS cloud profiles")
```
Comment (ialidzhikov) [inline pkg/admission/validator/cloudprofile.go:55]:
```suggestion
		return field.Required(providerConfigPath, "providerConfig must be set for Azure cloud profiles")
```

File: cmd/gardener-extension-admission-azure/app/app.go [hunk 3]
```diff
@@ -18,38 +18,42 @@ import (
...
 	controllercmd "github.com/gardener/gardener/extensions/pkg/controller/cmd"
+	webhookcmd "github.com/gardener/gardener/extensions/pkg/webhook/cmd"
 	"github.com/gardener/gardener/pkg/apis/core/install"
 	"github.com/spf13/cobra"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 var log = logf.Log.WithName("gardener-extension-validator-azure")
```
Comment (ialidzhikov) [inline cmd/gardener-extension-admission-azure/app/app.go:33]:
```suggestion
var log = logf.Log.WithName("gardener-extension-admission-azure")
```

**Review by ialidzhikov — COMMENTED** [review]
Minor nits

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**